opm/parser/eclipse/EclipseState/Schedule/Group/GuideRate.hpp

```diff
@@ -109,7 +111,19 @@ struct GRValState {
 
     using GRValPtr = std::unique_ptr<GRValState>;
 
+    typedef std::pair<Phase,std::string> pair;
+
+    struct pair_hash
+    {
+        template <class T1, class T2>
+        std::size_t operator() (const std::pair<T1, T2> &pair) const
+        {
+            return std::hash<T1>()(pair.first) ^ std::hash<T2>()(pair.second);
+        }
+    };
+
     std::unordered_map<std::string, GRValPtr> values;
+    std::unordered_map<pair, double, pair_hash> injection_group_values;
```

Comment:
I would prefer: `std::unordered_map<std::string, std::unordered_map<Phase, double>>`

Reply:
I know. I tried that as well. But I got lost somewhere around here: 
https://github.com/OPM/opm-simulators/blob/205c4839192d2aa5ced7d6cb88ccede78f246dcd/ebos/eclmpiserializer.hh#L273
Maybe @akva2 knows how to do the nested map serialization.

Reply:
OK I understand; I don't know how to handle that either.